src/http.cpp

```diff
@@ -860,7 +860,7 @@ Request::post()
 void
 Request::terminate(const asio::error_code& ec)
 {
-    if (finishing_.exchange(true))
+    if (finishing_ and finishing_.exchange(true))
```

Comment:
finishing_.exchange(true) already returns the previous value, but does it in a atomic way